ufm-test-services/metamask/tests/prometheusUtils.ts

```diff
@@ -0,0 +1,98 @@
+import 'dotenv/config'
+import { z } from 'zod'
+import { Counter, Pushgateway } from 'prom-client'
+
+const env = z
+  .object({
+    PROMETHEUS_SERVER_URL: z.string().url(),
+    PROMETHEUS_PUSHGATEWAY_URL: z.string().url(),
+  })
+  .parse(process.env)
+
+const txSuccessMetricName = 'metamask_tx_success'
+const txFailureMetricName = 'metamask_tx_failuree'
+
+const txSuccessCounter = new Counter({
+  name: txSuccessMetricName,
+  help: 'A counter signifying the number of successful transactions sent with Metamask since last failure',
+})
+const txFailureCounter = new Counter({
+  name: txFailureMetricName,
+  help: 'A counter signifying the number of failed transactions sent with Metamask since last successful transaction',
+})
+
+export const getMetamaskTxCounterValue = async (isSuccess: boolean) => {
+  const metricName = isSuccess ? txSuccessMetricName : txFailureMetricName
+  const prometheusMetricQuery = `${env.PROMETHEUS_SERVER_URL}/api/v1/query?query=${metricName}`
+
+  const response = await fetch(prometheusMetricQuery)
+  if (!response.ok)
+    throw new Error(`Failed to fetch metric from: ${prometheusMetricQuery}`)
```

Comment:
```suggestion
    throw new Error(`Failed to fetch metric from: ${prometheusMetricQuery}`)
```
console.error teh status code and metric

Reply:
status and response.message